app/src/main/java/org/mifos/selfserviceapp/ui/fragments/FundTransferFragment.java

```diff
@@ -0,0 +1,209 @@
+        super.onCreate(savedInstanceState);
+        ((BaseActivity) getActivity()).getActivityComponent().inject(this);
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        rootView = inflater.inflate(R.layout.fragment_fund_transfer, container, false);
+        ButterKnife.bind(this, rootView);
+
+        mFundTransferPresenter.attachView(this);
+
+        mFundTransferPresenter.loadAccountTransferTemplate();
+
+        return rootView;
+    }
+
+    @Override
+    public void showFundTransferTemplate(final FundTransferTemplateResponse fundTransferTemplate) {
+        List<FundTransferTemplate> fromAccountNumber = fundTransferTemplate.getFromAccountOptions();
+        List<FundTransferTemplate> toAccountNumber = fundTransferTemplate.getToAccountOptions();
+
+        String[] fromAccountItems = new String[fromAccountNumber.size()];
+        String[] toAccountItems = new String[toAccountNumber.size()];
+
+        for (int i = 0; i < fromAccountNumber.size(); i++) {
+            fromAccountItems[i] = fromAccountNumber.get(i).getAccountNo();
+        }
+        ArrayAdapter<String> fromAccountAdapter = new ArrayAdapter<>(getActivity(), android.R.layout.simple_list_item_1, fromAccountItems);
+        spFromAccount.setAdapter(fromAccountAdapter);
+        spFromAccount.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+            @Override
+            public void onItemSelected(AdapterView<?> parent, View view, int position, long
+                    id) {
+                fromClientId = fundTransferTemplate.getFromAccountOptions().
+                        get(position).getClientId();
+                fromAccountId = fundTransferTemplate.getFromAccountOptions().
+                        get(position).getAccountId();
+                fromAccountType = fundTransferTemplate.getFromAccountOptions().
+                        get(position).getAccountType().getId();
+                fromOfficeId = fundTransferTemplate.getFromAccountOptions().
+                        get(position).getOfficeId();
+            }
+
+            @Override
+            public void onNothingSelected(AdapterView<?> parent) {
+
+            }
+        });
+
+        for (int i = 0; i < toAccountNumber.size(); i++) {
+            toAccountItems[i] = toAccountNumber.get(i).getAccountNo();
+        }
+        ArrayAdapter<String> toAccountAdapter = new ArrayAdapter<>(getActivity(), android.R.layout.simple_list_item_1, toAccountItems);
+        spToAccount.setAdapter(toAccountAdapter);
+        spToAccount.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+            @Override
+            public void onItemSelected(AdapterView<?> parent, View view, int position, long
+                    id) {
+                toClientId = fundTransferTemplate.getToAccountOptions().
```

Comment:
same as above. Validate data else it might be a candidate for runtime exceptions.